### PR TITLE
prometheus: Remove stale worker metrics

### PR DESCRIPTION
**Pull request by databus23**
This fixes a bug reported in https://github.com/concourse/concourse/issues/2244

It removes gauge metrics for workers that haven't received a metric event in 5 minutes (because they are stalled or retired).

/cc @TimSimmons @jmcarp

### CODE REVIEW

**Comment by databus23** [comment]
I updated the PR to only handle stale worker metrics for now. Thinking about this some more I think I would rather change the scheduling metrics to counters which are not as problematic as gauges when the time series does not get anymore updates (because the pipeline is paused or deleted). I will create a separate PR for that.

File: metric/emitter/prometheus.go [hunk 1]
```diff
@@ -421,3 +435,33 @@ func (emitter *PrometheusEmitter) databaseMetrics(logger lager.Logger, event met
...
+func (emitter *PrometheusEmitter) periodicMetricGC() {
+	for {
+		emitter.mu.Lock()
+		now := time.Now()
+		for worker, lastSeen := range emitter.workerLastSeen {
+			if now.Sub(lastSeen) > 5*time.Minute {
+				//this is a little stupid but we don't know the platform here,
+				//so we try all known platoforms
+				for _, platform := range []string{"linux", "windows", "darwin"} {
+					emitter.workerContainers.DeleteLabelValues(worker, platform)
```
Comment (vito) [inline metric/emitter/prometheus.go]: Does this need the platform? Shouldn't the worker name be enough? Those are globally unique within Concourse's database.

Comment (databus23) [reply metric/emitter/prometheus.go]: The problem is that `DeleteLabelValues` only removes timeseries that have an exact match ok the label set. 

See https://godoc.org/github.com/prometheus/client_golang/prometheus#GaugeVec.DeleteLabelValues

Comment (vito) [reply metric/emitter/prometheus.go]: Gotcha, so it runs against all platforms just to cover all the bases. Wish there was a better way to do that, but it makes sense.

Unfortunately there are conflicts now so I'll hold off for a rebase, but I'll approve the changes for now.

**Comment by databus23** [comment]
@vito squashed and rebased.

**Comment by vito** [comment]
thanks!

**Pull request merged**